File: assets/problems/2023/cola/sol/subtask3.py
ceros = {}

pos = 1
salto = 2
while pos <= 10**9:
    ceros[pos] = True
    pos += salto
    salto += 1

N = int(input())
for _ in range(N):
    X = int(input())
    if ceros.get(X):
        print(0)
    else:
        print(1)
    

File: assets/problems/2023/cola/sol/subtask3.cpp
#include <iostream>
#include <map>

using namespace std;

int main() {
  map<int, bool> ceros;
  int pos = 1, salto = 2;
  while (pos <= 1000000000) {
    ceros[pos] = true;
    pos += salto;
    salto++;
  }
  int n, x;
  cin >> n;
  while (n--) {
    cin >> x;
    cout << (ceros[x] ? '0' : '1') << endl;
  }
  return 0;
}